src/shims/sync.rs

```diff
@@ -346,9 +346,12 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         if kind == this.eval_libc("PTHREAD_MUTEX_NORMAL")? {
             // In `glibc` implementation, the numeric values of
             // `PTHREAD_MUTEX_NORMAL` and `PTHREAD_MUTEX_DEFAULT` are equal, but
-            // they have different behaviour in some cases. Therefore, we add
-            // this flag to ensure that we can distinguish
-            // `PTHREAD_MUTEX_NORMAL` from `PTHREAD_MUTEX_DEFAULT`.
+            // they have different behaviour in some cases. Therefore, we assume
```

Comment:
I think it is wrong to say "PTHREAD_MUTEX_NORMAL and PTHREAD_MUTEX_DEFAULT have different behavior in some cases". That is not true, in any program you can replace one by the other and behavior is unchanged. So I think this comment is still confusing.

Reply:
What *does* have different behavior is *not setting anything* and *explicitly setting PTHREAD_MUTEX_DEFAULT*. That is the confusing point this comment should be all about.